### PR TITLE
Interact with API

### PR DESCRIPTION
**Pull request by piero-vic**
# Interact with API
- Add `fetchAddBook` and `fetchRemoveBook` to interact with API.
- Add `getBook` function to extract data from the API.
- Add action creator `populateList` to display items upon loading.

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -9,12 +9,13 @@ function Book(props) {
...
-    dispatch(removeBook(id));
+    dispatch(fetchRemoveBook(id));
   };
+
   return (
     <li>
-      {`${book.title} by ${book.author}`}
-      <button type="button" id={book.id} onClick={deleteBookFromStore}>Delete</button>
+      {`${book.title} | ${book.category} `}
+      <button type="button" id={book.item_id} onClick={deleteBookFromStore}>Delete</button>
```
Comment (cyonii) [inline src/components/Book.js:18]: - [ ] Please make sure the `deleteBook` feature works as expected. It should delete the book from the server and also remove it from the `localStorage`. Currently I cannot verify if it works as the `addBook` function is not working too.

File: src/components/BooksList.js [hunk 2]
```diff
@@ -1,44 +1,50 @@
...
+            <Book key={book.title} book={book} />
           ))
         }
       </ul>
       <form action="" onSubmit={submitBookToStore}>
         <input type="text" name="" placeholder="Book" id="books-input" required />
-        <input type="text" name="" placeholder="Author" id="author-input" required />
+        <input type="text" name="" placeholder="Category" id="category-input" required />
         <button type="submit">Add Book</button>
       </form>
```
Comment (cyonii) [inline src/components/BooksList.js:49]: - [ ] When I try to add book the server request returns an `HTTP 500` error, and the book is not added. You can look into the `fetchAddBook` function to make sure it is hitting the server as supposed.

File: src/components/BooksList.js [hunk 3]
```diff
@@ -1,44 +1,50 @@
...
+    categoryInput.value = '';
   };
 
   return (
     <div className={styles.container}>
       <ul>
         {
           booksStore.map((book) => (
-            <Book key={book.id} book={book} />
+            <Book key={book.title} book={book} />
```
Comment (cyonii) [inline src/components/BooksList.js:41]: - [ ] When your app loads initially this error is outputted in the browser console. Please make sure to assign "unique" key to each `Book` component.
Screen Shot 2021-12-08 at 7 23 40 PM

File: src/components/BooksList.js [hunk 4]
```diff
@@ -1,44 +1,50 @@
...
+import getBooks from '../getBooks';
 
 function BooksList() {
   const booksStore = useSelector((state) => state.books);
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    getBooks().then((result) => dispatch(populateList(result)));
+  }, []);
+
```
Comment (cyonii) [inline src/components/BooksList.js:16]: - [ ] When your app loads it makes a call to get books from the server but I noticed that this returns a `500`, which causes your app to display the data as `undefined`. You should look into the `getBooks` function and make sure it returns the expected result.
Screen Shot 2021-12-08 at 7 27 17 PM

**Review by cyonii — CHANGES_REQUESTED** [review]
# Changes Required ♻️ 

Hi @piero-vic, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes
_Check the comments under the review to see the changes you need to make._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BooksList.js (hunk 4 above)
Comment (piero-vic) [reply src/components/BooksList.js:16]: Hi. The app works fine on my side. You can see here that the url has `process.env.REACT_APP_API_KEY`. I think your not using the enviroment variable that I send in the submition.
https://github.com/piero-vic/bookstore/blob/fa1eeb8661f361067a5ddf44445c617a341c32fa/src/redux/books/books.js#L50
Is this okay or should I just hardcode it?

Comment (cyonii) [reply src/components/BooksList.js:16]: Hey @piero-vic, I see this is the issue; I apologize for the error.

You don't need to hardcore it, just point it for the next reviewer. And I hope they see this conversation.

**Comment by piero-vic** [comment]
For the next reviewer. Please don't forget to create a `.env` file in the project. I'm sending you the API key on the submition.

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @piero-vic, 

Everything works great :+1: 
Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sinansevgi** [comment]
> For the next reviewer. Please don't forget to create a `.env` file in the project. I'm sending you the API key on the submition.

You should include this in your readme file. Instructions in your readme file should be clear enough to run and use the project. Developing the skill of writing good documentation will improve your communication skills. Hence, you will explain even the most complex of concepts in a simple, straightforward way, which will help you in your interviews.

**Pull request merged**